kanidmd_web_ui/src/credential/totpmodal.rs

```diff
@@ -278,29 +279,36 @@ impl Component for TotpModalApp {
                 >{ "Accept SHA1 Token" }</button>
             },
             _ => html! {
-                <button id="totp-submit" type="button" class="btn btn-primary"
+                <button
+                    class="btn btn-primary"
                     disabled={ !submit_enabled }
+                    id="totp-submit"
                     onclick={
                         ctx.link()
-                            .callback(move |_| {
-                                Msg::TotpSubmit
-                            })
+                        .callback(move |_| {
+                            Msg::TotpSubmit
+                        })
                     }
+                    type="button"
                 >{ "Submit" }</button>
             },
         };
 
         let totp_secret_state = match &self.secret {
+            // TODO: the user's already clicked start, why are we asking them to press it again? :D
```

Comment:
Because we have to delay the fetch. The component is built on page load and hidden, so we have to wait for interaction to start a fetch and I'm too stoopid to work out how to make it "happen on display of the modal" so this dumb solution works.

Reply:
cool cool, I have ideas around this so will poke at it, having played with other bits recently ❤️